File: SoloLearn/It's a Sign/ItsaSign.java
import java.util.Scanner;

public class ItsaSign {

    public static String saveSign(Scanner input) {
        String label, reverse;
        for (int i=0; i<4; i++) {
            label = input.nextLine();
            reverse = new StringBuilder(label).reverse().toString();
            if (label.equals(reverse))
                return "Open";
        }
        return "Trash";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println(saveSign(scanner));
        scanner.close();
    }
}

File: SoloLearn/It's a Sign/ItsaSign.cpp
#include <iostream>
using namespace std;

string saveSign() {
    string label, reverse;
    for (int i=0; i<4; i++) {
        cin >> label;
        reverse = string(label.rbegin(), label.rend());
        if (label == reverse)
            return "Open";
    }
    return "Trash";
}

int main() {
    cout << saveSign();
    return 0;
} 